### PR TITLE
feat: basic filter support

### LINKED ISSUE #12: filter support
We want to be able to pass each event through a series of filter plugins in a pipeline. We should initially have an expression-based filter plugin, but we can also have plugins for things like executing an external command or executing a custom function (when using the Go API).

For UX reasons, we may want to not automatically register filter plugins and have an explicit config option for providing the filter expression via the CLI.

### PR DESCRIPTION
**Pull request by agaffney**
Fixes #12

### CODE REVIEW

**Comment by agaffney** [comment]
```
$ ./snek -filter-type chainsync.rollback
{"level":"info","timestamp":"2023-06-07T15:30:05-05:00","caller":"log/log.go:53","msg":"","type":"event","event":{"type":"chainsync.rollback","timestamp":"2023-06-07T15:30:05.414402313-05:00","payload":{"blockHash":"8d1a1dc757354d2883c986b2a8c9379df825f5be593fd9fff50339d17318b575","slotNumber":94603499}}}
^C

$ ./snek -filter-type chainsync.block
{"level":"info","timestamp":"2023-06-07T15:30:21-05:00","caller":"log/log.go:53","msg":"","type":"event","event":{"type":"chainsync.block","timestamp":"2023-06-07T15:30:21.262418828-05:00","payload":{"blockNumber":8874949,"blockHash":"1cfb379f0a128ca9310640a910c377a83aa304ab1e5f1bb9291732f2a8db965f","slotNumber":94603529}}}
^C

$ ./snek
{"level":"info","timestamp":"2023-06-07T15:30:29-05:00","caller":"log/log.go:53","msg":"","type":"event","event":{"type":"chainsync.rollback","timestamp":"2023-06-07T15:30:29.531732091-05:00","payload":{"blockHash":"1cfb379f0a128ca9310640a910c377a83aa304ab1e5f1bb9291732f2a8db965f","slotNumber":94603529}}}
{"level":"info","timestamp":"2023-06-07T15:31:05-05:00","caller":"log/log.go:53","msg":"","type":"event","event":{"type":"chainsync.block","timestamp":"2023-06-07T15:31:05.004821996-05:00","payload":{"blockNumber":8874950,"blockHash":"57f8c5815cffa32466dd12d019805ee08b665b0add81c7c3e1167a3b0c41b2bd","slotNumber":94603573}}}
{"level":"info","timestamp":"2023-06-07T15:31:05-05:00","caller":"log/log.go:53","msg":"","type":"event","event":{"type":"chainsync.transaction","timestamp":"2023-06-07T15:31:05.004833555-05:00","payload":{"blockNumber":8874950,"blockHash":"57f8c5815cffa32466dd12d019805ee08b665b0add81c7c3e1167a3b0c41b2bd","slotNumber":94603573,"transactionHash":"bc13869c7619d76fbcc5b535fc4763c067b5dd8fff00776f94f11386911175da"}}}
...
```

File: input/chainsync/chainsync.go [hunk 1]
```diff
@@ -202,10 +202,10 @@ func (c *ChainSync) handleRollForward(blockType uint, blockData interface{}, tip
 		if err != nil {
 			return err
 		}
-		blockEvt := event.New("block", time.Now(), NewBlockEvent(block, c.includeCbor))
+		blockEvt := event.New("chainsync.block", time.Now(), NewBlockEvent(block, c.includeCbor))
 		c.eventChan <- blockEvt
 		for _, transaction := range block.Transactions() {
-			txEvt := event.New("transaction", time.Now(), NewTransactionEvent(block, transaction, c.includeCbor))
+			txEvt := event.New("chainsync.transaction", time.Now(), NewTransactionEvent(block, transaction, c.includeCbor))
```
Comment (wolf31o2) [inline input/chainsync/chainsync.go:208]: Should this be block.transaction?

File: filter/event/event.go [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+				}
+			}
+			// Send event along
+			e.outputChan <- evt
+		}
+	}()
+	return nil
+}
+
+// Stop the chain sync filter
```
Comment (wolf31o2) [inline filter/event/event.go:64]: Isn't it a generic event filter?

**Review by wolf31o2 — APPROVED** [review]
Two questions, but neither are blocking

File: input/chainsync/chainsync.go (hunk 1 above)
Comment (agaffney) [reply input/chainsync/chainsync.go:208]: No, it's meant to be `<plugin>.<type>`, but we may want to make adding that `<plugin>` prefix automatic

File: filter/event/event.go (hunk 2 above)
Comment (agaffney) [reply filter/event/event.go:64]: Copy-pasta fail

**Review by wolf31o2 — APPROVED** [review]

**Pull request merged**